### DISCUSSION TITLE
Extend Gattling logic to infantry and aircraft

### DISCUSSION THREAD

**Discussion by MortonPL** [Phobos-developers/Phobos#878] [Ideas and feedback]
As of now, infantry and pad-bound aircraft TechnoTypes cannot use IsGattling=yes and all of its additions. I propose to extend the Gattling weapon logic to work with all unit TechnoTypes, not only vehicles and defences. For infantry, no special changes to this logic would be necessary. For pad-bound aircraft, Gattling logic would need to take limited, cant-reload-in-flight ammo into consideration.

**Comment by Musketmarine** [comment]
This sounds incredible, i'd like to see this

**Comment by GraionDilach** [comment]
> For infantry, no special changes to this logic would be necessary. For pad-bound aircraft, Gattling logic would need to take limited, cant-reload-in-flight ammo into consideration.

The Gattling state variables only exist in `VehicleTypes`. In order to do this, one needs to move these variables out to `TechnoTypeExt`.
